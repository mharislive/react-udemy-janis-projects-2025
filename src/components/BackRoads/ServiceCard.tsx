import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import type { IconDefinition } from "@fortawesome/fontawesome-svg-core";

type ServiceCardProps = {
  icon: IconDefinition;
  title: string;
  content: string;
};

const ServiceCard = (props: ServiceCardProps) => {
  const { content, title, icon } = props;

  return (
    <article className="flex gap-4 flex-1">
      <div className="p-2 bg-cyan-500 text-cyan-900 text-4xl self-start">
        <FontAwesomeIcon icon={icon} aria-label={title} data-testid="icon" />
      </div>
      <div>
        <h3 className="capitalize tracking-widest font-bold mb-2">{title}</h3>
        <p className="text-slate-500">{content}</p>
      </div>
    </article>
  );
};

export default ServiceCard;
