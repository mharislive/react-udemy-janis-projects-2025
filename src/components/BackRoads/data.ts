import { faSocks, faTree, faWallet } from "@fortawesome/free-solid-svg-icons";
import tour1 from "../../assets/backroads/tour-1.jpeg";
import tour2 from "../../assets/backroads/tour-2.jpeg";
import tour3 from "../../assets/backroads/tour-3.jpeg";
import tour4 from "../../assets/backroads/tour-4.jpeg";

export const serviceData = [
  {
    title: "saving money",
    content:
      "Lorem ipsum dolor sit amet consectetur adipisicing elit. Asperiores, officia.",
    icon: faWallet,
  },
  {
    title: "endless hiking",
    content:
      "Lorem ipsum dolor sit amet consectetur adipisicing elit. Asperiores, officia.",
    icon: faTree,
  },
  {
    title: "amazing comfort",
    content:
      "Lorem ipsum dolor sit amet consectetur adipisicing elit. Asperiores, officia.",
    icon: faSocks,
  },
];

export const toursdata = [
  {
    content:
      "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.",
    image: tour1,
    date: "august 26th, 2020",
    title: "Tibet Adventure",
    country: "china",
    duration: "6 days",
    price: "$2100",
  },
  {
    content:
      "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.",
    image: tour2,
    date: "october 1th, 2020",
    title: "best of java",
    country: "indonesia",
    duration: "11 days",
    price: "$1400",
  },
  {
    content:
      "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.",
    image: tour3,
    date: "september 15th, 2020",
    title: "explore hong kong",
    country: "hong kong",
    duration: "8 days",
    price: "$5000",
  },
  {
    content:
      "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cumque vitae tempore voluptatum maxime reprehenderit eum quod exercitationem fugit, qui corporis.",
    image: tour4,
    date: "december 5th, 2019",
    title: "kenya highlights",
    country: "kenya",
    duration: "20 days",
    price: "$3300",
  },
];
