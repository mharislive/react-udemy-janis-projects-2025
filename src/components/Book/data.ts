import type { BookType } from "../../types/Book";

export const BOOKS: BookType[] = [
  {
    title: "Atomic Habits",
    author: "James Clear",
    image: "https://m.media-amazon.com/images/I/81F90H7hnML._SY425_.jpg",
    id: 1,
  },
  {
    title: "Word Power Made Easy",
    author: "Norman Lewis",
    image: "https://m.media-amazon.com/images/I/71DortY2MXL._SY425_.jpg",
    id: 2,
  },
  {
    title: "The Psychology of Money Paperback",
    author: "Morgan Housel",
    image: "https://m.media-amazon.com/images/I/71XEsXS5RlL._SY425_.jpg",
    id: 3,
  },
];
