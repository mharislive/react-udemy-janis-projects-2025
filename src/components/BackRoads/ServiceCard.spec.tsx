import { render, screen } from "@testing-library/react";
import { axe } from "vitest-axe";
import ServiceCard from "./ServiceCard";
import { serviceData } from "./data";

const service = serviceData[0];

describe("BackRoads - Service Card", () => {
  it("should render successfully", () => {
    render(<ServiceCard {...service} />);
    expect(screen.getByRole("article")).toBeInTheDocument();
  });

  it("should render icon", () => {
    render(<ServiceCard {...service} />);
    expect(screen.getByLabelText(service.title)).toBeInTheDocument();
  });

  it("should render title", () => {
    render(<ServiceCard {...service} />);
    expect(screen.getByRole("heading", { name: service.title })).toBeVisible();
  });

  it("should render content", () => {
    render(<ServiceCard {...service} />);
    expect(screen.getByRole("paragraph")).toBeVisible();
    expect(screen.getByText(/lorem/i)).toBeVisible();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(<ServiceCard {...service} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should have no accessibility violations", async () => {
    let { container } = render(<ServiceCard {...service} />);
    let results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
