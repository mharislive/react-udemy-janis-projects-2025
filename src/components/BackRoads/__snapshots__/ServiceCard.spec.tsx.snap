// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BackRoads - Service Card > should match snapshot 1`] = `
<DocumentFragment>
  <article
    class="flex gap-4 flex-1"
  >
    <div
      class="p-2 bg-cyan-500 text-cyan-900 text-4xl self-start"
    >
      <svg
        aria-hidden="true"
        aria-label="saving money"
        class="svg-inline--fa fa-wallet "
        data-icon="wallet"
        data-prefix="fas"
        data-testid="icon"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M64 32C28.7 32 0 60.7 0 96L0 416c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-224c0-35.3-28.7-64-64-64L80 128c-8.8 0-16-7.2-16-16s7.2-16 16-16l368 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L64 32zM416 272a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div>
      <h3
        class="capitalize tracking-widest font-bold mb-2"
      >
        saving money
      </h3>
      <p
        class="text-slate-500"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Asperiores, officia.
      </p>
    </div>
  </article>
</DocumentFragment>
`;
